torch_geometric/datasets/generators/graph_generator.py

```diff
@@ -0,0 +1,84 @@
+from abc import abstractmethod
+from typing import Callable, Optional
+
+import torch
+
+from torch_geometric.data import Data
+from torch_geometric.seed import seed_everything
+
+
+class GraphGenerator:
+    r"""Base class for generating benchmark datasets. It contains
+        `generate_feature` and `attach_motif` methods used to generate
+        features and attach motifs to base graph. The motifs are
+        currently attached in random order.
+
+    Args:
+        num_nodes (int): The number of nodes used to attach the motifs.
+        motif (:obj:`toch_geometric.datasets.generators.Motif`, Optional):
+            Motif object to be attached to the base graph.
+        seed (int, Optional): seed number for the generator.
+    """
+    def __init__(self, num_nodes: int = 300, motif: Optional[Callable] = None,
+                 seed: int = None):
+        self.num_nodes = num_nodes
+        self.motif = motif
+        self.seed = seed
+        self._edge_index = None
+        self._edge_label = None
+        self._expl_mask = None
+        self._node_label = None
+        self._x = None
+
+    r"""Abstracted method to be implemented by a specific GraphGenerator.
+    """
+
+    @abstractmethod
+    def generate_base_graph(self) -> Data:
+        raise NotImplementedError
+
+    def generate_graph(self):
+        if self.seed:
+            seed_everything(self.seed)
+        self.generate_base_graph()
+
+    @property
+    def data(self):
+        return Data(x=self._x, edge_index=self._edge_index, y=self._node_label,
```

Comment:
Given that `data(self)` returns a `Data` class with an additional attribute `expl_mask`, it would make more sense to return an `Explanation` here. WDYT? 

We have created the `Explanation` class just for this purpose, to hold explanation masks (of different kinds, node mask in this case) along with other  `Data` attributes. This also makes the interface to ground truth explanation evaluation more clear (which is the purpose of these datasets anyway), e.g. we are really comparing two explanations there `gt_metics(explanation: Explanation, groundtruth: Explanation)`.

Reply:
Thank you @BlazStojanovic  for bringing up those relevant points!
Since Explanation holds specific properties that Data is already returning here, as you pointed out, it makes sense. 
Is there any use case where GraphGenerator would be used to generate synthetic graphs not for explainability but for another purpose?

Reply:
I don't thinks so, and even if there was another use, the Explanation is still a `Data` subclass, and can be used as one.

Reply:
PR updated, thanks @BlazStojanovic